service/tablet_allocator.cc

```diff
@@ -1149,6 +1150,36 @@ class load_balancer {
     }
 };
 
+future<tablet_reallocation_result>
+reallocate_tablets_for_new_rf(const tablet_aware_replication_strategy* rep, schema_ptr s, token_metadata_ptr tm,
+    std::unordered_map<sstring, sstring> const& ks_options) {
+
+    auto* net_rep = dynamic_cast<const locator::network_topology_strategy*>(rep);
+
+    auto table_id = s->id();
+    tablet_map old_tablets = tm->tablets().get_tablet_map(table_id);
+
+    if (!net_rep) {
+        co_return tablet_reallocation_result{ old_tablets, tablet_reallocation_status::unknown_topology_strategy };
+    }
```

Comment:
Casting repl strategy and this `if` should be combined under a single if and moved to the top of this function.

Reply:
OK, I can see now why it's done this way.